tico/experimental/quantization/passes/insert_quantize_on_dtype_mismatch.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import torch.fx
+import copy
+
+import torch
+from torch.export import ExportedProgram
+
+from tico.serialize.quant_param import QPARAM_KEY, QuantParam
+from tico.utils import logging
+from tico.utils.errors import NotYetSupportedError
+from tico.utils.passes import PassBase, PassResult
+from tico.utils.trace_decorators import trace_graph_diff_on_pass
+from tico.utils.utils import set_new_meta_val
+from tico.utils.validate_args_kwargs import (
+    BmmArgs,
+    LinearArgs,
+    MulTensorArgs,
+    PermuteArgs,
+)
+
+
+def _dtype(node: torch.fx.Node) -> str:
+    assert QPARAM_KEY in node.meta
+    return node.meta[QPARAM_KEY].dtype
+
+
+# Convert i16 qparam to u8 qparam
+# scale and zero_point are inferred from i16 qparam
+def _i16_to_u8(qparam: QuantParam) -> QuantParam:
+    # Assume per-tensor quantization
+    assert qparam.scale is not None and len(qparam.scale) == 1
+    assert qparam.dtype == "int16"
+
+    s16_scale = qparam.scale[0]
+    max_ = s16_scale * 32767  # numeric_limits<int16>
+    min_ = -max_
+
+    u8_scale = (max_ - min_) / 255
+    u8_zerop = round(-min_ / u8_scale)
+
+    new_qparam = QuantParam()
+    new_qparam.scale = [u8_scale]
+    new_qparam.zero_point = [u8_zerop]
+    new_qparam.dtype = "uint8"
+
+    return new_qparam
+
+
+# Convert u8 qparam to i16 qparam
+# scale is inferred from u8 qparam
+def _u8_to_i16(qparam: QuantParam) -> QuantParam:
+    # Assume per-tensor quantization
+    assert qparam.scale is not None and len(qparam.scale) == 1
+    assert qparam.zero_point is not None and len(qparam.zero_point) == 1
+    assert qparam.dtype == "uint8"
+
+    u8_scale = qparam.scale[0]
+    u8_zerop = qparam.zero_point[0]
+    max_ = u8_scale * (255 - u8_zerop)
+    min_ = u8_scale * (-u8_zerop)
+
+    abs_max = max([max_, min_], key=abs)
+    s16_scale = abs_max / 32767
+    s16_zerop = 0
+
+    new_qparam = QuantParam()
```

Comment:
`.min` and `.max` also should be saved.

Reply:
Is it necessary? Typically they are not used.

FYI one-quantize removes min/max by default to reduce model size.